src/xcode/ENA/ENA/Source/Scenes/QRScanner/Model/Parser/CheckinQRCodeParser.swift

```diff
@@ -57,7 +57,7 @@ class CheckinQRCodeParser: QRCodeParsable {
 		onSuccess: @escaping((TraceLocation) -> Void),
 		onError: @escaping((CheckinQRScannerError) -> Void)
 	) {
-		appConfigurationProvider.appConfiguration().sink { [weak self] appConfig in
+		appConfigurationProvider.currentAppConfig.sink { [weak self] appConfig in
```

Comment:
An issue in both of this cases could be that as long as they are in memory the `currentAppConfig` publisher could publish again and again and the completion handlers could be called multiple times which then could lead to unexpected behavior. I guess it would be safer to just synchronously access the current app config once.